src/index.ts

```diff
@@ -0,0 +1,189 @@
+    window._fs_run_in_iframe = true;
+  }
+
+  // record the contents of this iFrame when embedded in a parent site
+  if (options.recordOnlyThisIFrame) {
+    window._fs_is_outer_script = true;
+  }
+
+  // Set cookie domain if it was specified.
+  if (options.cookieDomain) {
+    window._fs_cookie_domain = options.cookieDomain;
+  }
+
+  if (options.debug === true) {
+    if (!options.script) {
+      options.script = 'edge.fullstory.com/s/fs-debug.js';
+    } else {
+      console.warn('Ignoring `debug = true` because `script` is set');
+    }
+  }
+
+  initFS(options);
+
+  const fs = getFullStory();
+
+  if (!fs) {
+    console.warn('Trouble with fs');
+    return;
+  }
+
+  if (readyCallback) {
+    fs('observe', { type: 'start', callback: readyCallback });
+  }
+
+  if (options.devMode === true) {
+    const message = 'FullStory was initialized in devMode and will stop recording';
+    fs('trackEvent', {
+      name: 'FullStory Dev Mode',
+      properties: {
+        message_str: message,
+      }
+    });
+    fs('shutdown');
+    window._fs_dev_mode = true;
+    console.warn(message);
+  }
+};
+
+const initOnce = (message) => (inputOptions: SnippetOptions, readyCallback?: ReadyCallback) => {
+  if (window._fs_initialized) {
+    if (message) console.warn(message);
+    return;
+  }
+  _init(inputOptions, readyCallback);
+  window._fs_initialized = true;
+};
+
+export const init = initOnce('FullStory init has already been called once, additional invocations are ignored');
+
+// normalize undefined into boolean
+export const isInitialized = () => !!window._fs_initialized;
+
+const hasFullStoryWithFunction = (...testNames) => {
+  const fs = ensureSnippetLoaded();
+  return testNames.every((current) => fs[current]);
+};
+
+const guard = (name) => (...args) => {
+  if (window._fs_dev_mode) {
+    const message = `FullStory is in dev mode and is not recording: ${name} method not executed`;
+    console.warn(message);
+    return message;
+  }
+
+  const fs = getFullStory();
+  if (hasFullStoryWithFunction(name) && fs) {
+    return fs[name](...args);
+  }
+  console.warn(`FS.${name} not ready`);
+  return null;
+};
+
+const FullStory: FSApi = Object.assign(
+  (operation, options, source) => {
+    const fs = ensureSnippetLoaded();
+
+    if (window._fs_dev_mode) {
+      const message = 'FullStory is in dev mode and is not recording: method not executed';
+      console.warn(message);
+      return undefined;
+    }
+
+    return fs(operation, options, source);
+  },
+  {
+    anonymize: guard('anonymize'),
+    consent: guard('consent'),
+    disableConsole: guard('disableConsole'),
+    enableConsole: guard('enableConsole'),
+    event: guard('event'),
+    getCurrentSessionURL: guard('getCurrentSessionURL'),
+    identify: guard('identify'),
+    log: guard('log'),
+    restart: guard('restart'),
+    setUserVars: guard('setUserVars'),
+    setVars: guard('setVars'),
+    shutdown: guard('shutdown'),
+  },
+) as FSApi;
```

Comment:
this is the `Object.assign` in question. It could just as easily be:
```js
const FullStory: FSApi = (operation, options, source) => {
  const fs = ensureSnippetLoaded();

  if (window._fs_dev_mode) {
    const message = 'FullStory is in dev mode and is not recording: method not executed';
    console.warn(message);
    return undefined;
  }

  return fs(operation, options, source);
};

FullStory.anonymize = guard('anonymize');
// etc...
```

I think you have to get creative about type declaration, but we could probably make it work. The `Object.assign` approach is arguably cleaner.